src/Mailer/Mailer.php

```diff
@@ -244,34 +313,189 @@ public function send(string $action, array $args = [], array $headers = []): arr
+        ];
+        foreach ($viewBuilderMethods as $method) {
+            if (array_key_exists($method, $config)) {
+                $this->viewBuilder()->{'set' . ucfirst($method)}($config[$method]);
+                unset($config[$method]);
+            }
+        }
+
+        if (array_key_exists('helpers', $config)) {
+            $this->viewBuilder()->setHelpers($config['helpers'], false);
+            unset($config['helpers']);
+        }
+        if (array_key_exists('viewRenderer', $config)) {
+            $this->viewBuilder()->setClassName($config['viewRenderer']);
+            unset($config['viewRenderer']);
+        }
+        if (array_key_exists('viewVars', $config)) {
+            $this->viewBuilder()->setVars($config['viewVars']);
+            unset($config['viewVars']);
+        }
+
+        $this->message->setConfig($config);
+
+        return $this;
+    }
+
+    /**
+     * Sets the transport.
+     *
+     * When setting the transport you can either use the name
+     * of a configured transport or supply a constructed transport.
+     *
+     * @param string|\Cake\Mailer\AbstractTransport $name Either the name of a configured
+     *   transport, or a transport instance.
+     * @return $this
+     * @throws \LogicException When the chosen transport lacks a send method.
+     * @throws \InvalidArgumentException When $name is neither a string nor an object.
+     */
+    public function setTransport($name)
+    {
+        if (is_string($name)) {
+            $transport = TransportFactory::get($name);
+        } elseif (is_object($name)) {
+            $transport = $name;
+            if (!$transport instanceof AbstractTransport) {
+                throw new Exception('Transport class must extend Cake\Mailer\AbstractTransport');
+            }
+        } else {
+            throw new InvalidArgumentException(sprintf(
+                'The value passed for the "$name" argument must be either a string, or an object, %s given.',
+                gettype($name)
+            ));
+        }
+
+        $this->transport = $transport;
+
+        return $this;
+    }
+
+    /**
+     * Gets the transport.
+     *
+     * @return \Cake\Mailer\AbstractTransport
+     */
+    public function getTransport(): AbstractTransport
+    {
+        if ($this->transport === null) {
+            throw new BadMethodCallException(
+                'Transport was not defined. You must set on using setTransport() or set `transport` option in profile.'
+            );
+        }
+
+        return $this->transport;
+    }
+
+    /**
+     * Restore message, renderer, transport instances to state before an action was run.
+     *
+     * @return $this
+     */
+    protected function restore()
+    {
+        foreach (array_keys($this->clonedInstances) as $key) {
+            if ($this->clonedInstances[$key] === null) {
+                $this->{$key} = null;
+            } else {
+                $this->{$key} = clone $this->clonedInstances[$key];
```

Comment:
Should the clonedInstances be cleared after all properties have been restored?

Reply:
Done